src/eliater/regression.py

```diff
@@ -0,0 +1,386 @@
+is the direct effect X -> Y.
+
+.. warning::
+
+    But there is an easier way to do it!  If you include a set of variables
+    that block all backdoor paths between X and Y in the original regression,
+    then the Beta coefficient associated with X will be the direct effect.
+    But be careful! If you include extra variables in the regression that open a
+    backdoor path between X and Y, then the Beta regression coefficient associated
+    with X will no longer represent the direct effect
+"""
+
+import statistics
+from operator import attrgetter
+from typing import Dict, Literal, NamedTuple, Optional, Sequence, Tuple
+
+import networkx.exception
+import pandas as pd
+from sklearn.linear_model import LinearRegression
+
+from eliater.io import to_bayesian_network, to_causal_graph
+from y0.dsl import Variable
+from y0.graph import NxMixedGraph, _ensure_set
+
+__all__ = [
+    # High-level functions
+    "estimate_query",
+    "estimate_ate",
+    "estimate_probabilities",
+    # Helper functions
+    "get_regression_results",
+    "get_adjustment_sets",
+    "fit_regressions",
+]
+
+
+Impl = Literal["pgmpy", "optimaladj"]
+
+
+class RegressionResult(NamedTuple):
+    """Represents a regression."""
+
+    coefficients: dict[Variable, float]
+    intercept: float
+
+
+def estimate_ate(
+    graph: NxMixedGraph,
+    data: pd.DataFrame,
+    treatment: Variable,
+    outcome: Variable,
+    *,
+    impl: Optional[Impl] = None,
+    conditions: None | Variable | set[Variable] = None,
+) -> float:
+    """Return a simplified view for a simplified regression scenario.
+
+    1. Gets all adjustment sets returned by the implementation. If using :mod:`pgmpy`, then
+       returns multiple. The shortest is non-deterministically chosen. If using  :mod:`optimaladj`,
+       then a single optimal set is chosen (though might error if not possible)
+    2. Fits a linear regression between the union of the treatment and variables in the adjustment
+       set against the outcome
+    3. Returns the coefficient for the treatment in the linear regression, which represents the direct
+       effect of the treatment on the outcome
+
+    :param graph: An acyclic directed mixed graph (ADMG)
+    :param data: Observational data corresponding to the ADMG
+    :param treatment: The treatment variable
+    :param outcome: The outcome variable
+    :param conditions: Conditions to apply to the query
+    :param impl: The implementation for getting adjustment sets.
+    :returns: The coefficient for the treatment in the linear regression between
+        the union of the treatment and (optimal/chosen) adjustment set and the outcome
+        as the response
+    """
+    _adjustment_set, results = _get_regression_result(
+        graph=graph,
+        data=data,
+        treatments=treatment,
+        outcome=outcome,
+        impl=impl,
+        conditions=conditions,
+    )
+    # TODO how would you aggregate the coefficients on multiple treatments?
```

Comment:
@djinnome it seems like it wouldn't be hard to extend the notion of using the treatment's coefficient as the direct effect for the ATE to have multiple treatments. If have multiple treatments, can we take the average of their coefficients? or do some other kind of operation?